cibuildwheel/ios.py

```diff
@@ -0,0 +1,589 @@
+
+    venv_path = tmp / "venv"
+    env = cross_virtualenv(
+        py_version=python_configuration.version,
+        host_python=host_python,
+        multiarch=python_configuration.multiarch,
+        build_python=build_python,
+        venv_path=venv_path,
+        dependency_constraint_flags=dependency_constraint_flags,
+    )
+    venv_bin_path = venv_path / "bin"
+    assert venv_bin_path.exists()
+
+    # We version pip ourselves, so we don't care about pip version checking
+    env["PIP_DISABLE_PIP_VERSION_CHECK"] = "1"
+
+    # upgrade pip to the version matching our constraints
+    # if necessary, reinstall it to ensure that it's available on PATH as 'pip'
+    pip = ["python", "-m", "pip"]
+    call(
+        *pip,
+        "install",
+        "--upgrade",
+        "pip",
+        *dependency_constraint_flags,
+        env=env,
+        cwd=venv_path,
+    )
+
+    # Apply our environment after pip is ready
+    env = environment.as_dictionary(prev_environment=env)
+
+    # Check what Python version we're on
+    which_python = call("which", "python", env=env, capture_stdout=True).strip()
+    print(which_python)
+    if which_python != str(venv_bin_path / "python"):
+        msg = (
+            "cibuildwheel: python available on PATH doesn't match our installed instance. "
+            "If you have modified PATH, ensure that you don't overwrite cibuildwheel's "
+            "entry or insert python above it."
+        )
+        raise errors.FatalError(msg)
+    call("python", "--version", env=env)
+
+    # Check what pip version we're on
+    assert (venv_bin_path / "pip").exists()
+    which_pip = call("which", "pip", env=env, capture_stdout=True).strip()
+    print(which_pip)
+    if which_pip != str(venv_bin_path / "pip"):
+        msg = (
+            "cibuildwheel: pip available on PATH doesn't match our installed instance. "
+            "If you have modified PATH, ensure that you don't overwrite cibuildwheel's "
+            "entry or insert pip above it."
+        )
+        raise errors.FatalError(msg)
+    call("pip", "--version", env=env)
+
+    # Ensure that IPHONEOS_DEPLOYMENT_TARGET is set in the environment
+    ios_deployment_target = os.getenv("IPHONEOS_DEPLOYMENT_TARGET", "13.0")
+    env["IPHONEOS_DEPLOYMENT_TARGET"] = ios_deployment_target
+
+    log.step("Installing build tools...")
+    if build_frontend == "pip":
+        # No additional build tools required
+        pass
+    elif build_frontend == "build":
+        call(
+            "pip",
+            "install",
+            "--upgrade",
+            "build[virtualenv]",
+            *dependency_constraint_flags,
+            env=env,
+        )
+    else:
+        assert_never(build_frontend)
+
+    return host_install_path, env
+
+
+def build(options: Options, tmp_path: Path) -> None:
+    if sys.platform != "darwin":
+        msg = "iOS binaries can only be built on macOS"
+        raise ValueError(msg)
```

Comment:
This will give a better formatted error message.

```suggestion
        raise errors.FatalError(msg)
```
(import from `from cibuildwheel import errors`)